### PR TITLE
Issue 876: add draft specification for role label

### PR DESCRIPTION
**Pull request by jongund**
***
<a href="https://pr-preview.s3.amazonaws.com/jongund/aria/pull/886.html" title="Last updated on Apr 4, 2019, 7:05 PM UTC (483e30b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria/886/57567af...jongund:483e30b.html" title="Last updated on Apr 4, 2019, 7:05 PM UTC (483e30b)">Diff</a>

### CODE REVIEW

**Comment by jnurthen** [comment]
#876

**Comment by jongund** [comment]
@jnurthen 
Updated the list of roles that can have name generated by label encapsulation based on discussions on the aria 2/21/2019 teleconference.
I also added "label encapsulation" to the list of "name from" for the included roles.

File: index.html [hunk 1]
```diff
@@ -3221,6 +3222,133 @@ <h2>Definition of Roles</h2>
 				</tbody>
 			</table>
 		</div>
+    <div class="role" id="label">
+      <rdef>label</rdef>
+      <div class="role-description">
+        <p>The <rref>label</rref> role can be used to identify the visible caption (e.g. acccessible name) of a user interface component. A user interface component can be associated with a label using the <code>aria-labelledby</code> attribute or by encapsulating the component inside the element with <code>role=label</code>.  Not all roles that support name from author can use the encapsualtion method, the following list are the roles that are supported: </p>
```
[the remark this answers is not in the document]
Comment (jnurthen) [reply index.html]: accccccessible can never have enough c's

**Comment by devarshipant** [comment]
@jongund - I don't think using role label provides the semantic equivalence of matching `for / id `attributes in native html. In that case, should there be instructions to warn authors not to rely on role label to set keyboard focus on form control by clicking the label?

**Comment by jongund** [comment]
@jnurthen 
I made some edits to fix spelling problem and improved the definition text that is used in Section 5.4 Definition of Roles.  So it should be ready to merge.

**Comment by jongund** [comment]
> @jongund - I don't think using role label provides the semantic equivalence of matching `for / id `attributes in native html. In that case, should there be instructions to warn authors not to rely on role label to set keyboard focus on form control by clicking the label?

There is an author directive to add this behavior when then are using the label role

**Comment by jongund** [comment]
@jnurthen @mcking65 
I made updates to the pull request based on your comments.

File: index.html [hunk 2]
```diff
@@ -530,6 +530,7 @@ <h3>Accessible Name Calculation</h3>
...
 						<li>contents: name comes from the text value of the <a>element</a> node. Although this may be allowed in addition to "author" in some <a>roles</a>, this is used in content only if higher priority "author" features are not provided. Priority is defined by the <a href="https://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te">accessible name and description computation</a> algorithm.  [[ACCNAME-1.1]]</li>
+            <li>encapsulation: name comes from the text value of the <a>element</a> node with role <cod>label</cod> that is the closest ancestor. Although this may be allowed in addition to "author" and "contents" in some <a>roles</a>, encapsulation is used only if higher priority "author" features are not provided. Priority is defined by the <a href="https://www.w3.org/TR/accname-aam-1.1/#mapping_additional_nd_te">accessible name and description computation</a> algorithm.  [[ACCNAME-1.1]]</li>
```
[the remark this answers is not in the document]
Comment (jnurthen) [reply index.html]: @jongund I think this comment still needs to be resolved before I can merge.

File: index.html [hunk 3]
```diff
@@ -3221,6 +3224,117 @@ <h2>Definition of Roles</h2>
 				</tbody>
 			</table>
 		</div>
+    <div class="role" id="label">
+      <rdef>label</rdef>
+      <div class="role-description">
+        <p>A visible name or caption for a user interface component.</p>
+        <p>A label element can provide an accessible name for a user interface component if it is programatically associated with the element. Authors MAY associate a label element with the element by using one of two methods:</p>
```
Comment (melanierichards) [inline index.html]: Wordsmithing: instead of "a label element", perhaps "An element with role `label`" (that said, we do seem to do both).

**Comment by jongund** [comment]
This pull request can be discarded, see new pull request

**Pull request closed without merging**